### PR TITLE
[WIP] Fix memset, memcpy, memmove calls on Pointer type.

### PR DESCRIPTION
**Pull request by akzhan**
Just note that it's short fix for #4589 only.

No generic SizeT, OffsetT introduced here.

### CODE REVIEW

File: src/pointer.cr [hunk 1]
```diff
@@ -129,6 +129,16 @@ struct Pointer(T)
...
+  {% if flag?(:x86_64) %}
+    private def to_size_t(size : Int) : UInt64
+        size.to_u64
+    end
+  {% else %}
+    private def to_size_t(size : Int) : UInt32
+        size.to_u32
+    end
+  {% end %}
+
```
Comment (oprypin) [inline src/pointer.cr]: Not needed. `LibC::SizeT.new`

Comment (akzhan) [inline src/pointer.cr]: No. `Intrinsics` method signatures doesn't related to `LibC::SizeT`.

This PR is short one, without changing of rules of the game.

Comment (oprypin) [inline src/pointer.cr]: No idea what you're talking about. Why spray the definition of size_t all over the codebase when it already exists.
Use `LibC::SizeT.new` instead of `to_size_t` and you don't need this method.

Comment (akzhan) [inline src/pointer.cr]: Yes, we need to introduce generic SizeT and OffsetT types, or change `Intrinsics` contracts, but it should be discussed and introduced by another PR.

This PR just fixes contract implementation.

File: src/pointer.cr [hunk 2]
```diff
@@ -242,7 +242,7 @@ struct Pointer(T)
     raise ArgumentError.new("Negative count") if count < 0
 
     if self.class == source.class
-      Intrinsics.memcpy(self.as(Void*), source.as(Void*), (count * sizeof(T)).to_u32, 0_u32, false)
+      Intrinsics.memcpy(self.as(Void*), source.as(Void*), count * sizeof(T), 0_u32, false)
```
Comment (RX14) [inline src/pointer.cr]: I have no idea why the CI isn't broken on 32bits but isn't this passing 32bit ints to something expecting a `U64`?

**Comment by zatherz** [comment]
Dislike this as it actually returns a different type depending on the architecture. I'd prefer a SizeT.

**Comment by akzhan** [comment]
This is a palliative. Will rework using SizeT and OffsetT.

**Comment by ysbaddaden** [comment]
Sorry, that doesn't solve anything. It still relies on `count` type, which is likely an Int32, then multiply by another int (whatever its type) that may overflow.

**Comment by konovod** [comment]
So it should be `SizeT.new(count) * sizeof(T)` ? Then count is still 32-bit, and result is SizeT (or whatever type will be choosen that match intrinsic param).

**Comment by akzhan** [comment]
@konovod thanks. was fixed with #4624.
